catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/MigrationFileWriter.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.migratable.impl;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.nio.file.Path;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.codice.ddf.migration.MigrationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Result;
+
+/**
+ * Responsible for all low-level IO operations pertaining to migration.
+ */
+public class MigrationFileWriter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MigrationFileWriter.class);
+
+    //      ========================================================
+    //      TODO: Look into moving these functions to MigratableUtil
+    //      ========================================================
+
+    /**
+     * Creates the directory for the catalog export or if it already exists, deletes it and
+     * recreates it.
+     *
+     * @param exportPath The path representing the directory to create.
+     * @throws MigrationException thrown if the export directory could not be created or emptied
+     */
+    public void init(Path exportPath) throws MigrationException {
+        final File exportDir = exportPath.toFile();
+        if (!exportDir.exists() || !exportDir.isDirectory()) {
+            if (!exportDir.mkdir()) {
+                throw new MigrationException("Could not create catalog directory at export location");
+            } else {
+                LOGGER.info("Created export directory: 'catalog'");
+            }
+        } else {
+            try {
+                FileUtils.cleanDirectory(exportDir);
```

Comment:
Great minds think alike :-) We did create a ticket for this, which means that this code will be removed from this class.